addons/binding/org.openhab.binding.autelis/src/main/java/org/openhab/binding/autelis/handler/AutelisHandler.java

```diff
@@ -0,0 +1,401 @@
+		password = configuration.pass;
+		refresh = configuration.refresh.intValue();
+		
+		baseURL = "http://" + host + ":" + port;
+
+		logger.debug("Autelius binding configured with base url {}", baseURL);
+
+		properlyConfigured = true;
+
+		initPolling();
+	}
+
+	/**
+	 * starts this things polling future
+	 */
+	private void initPolling() {
+		stopPolling();
+		pollFuture = scheduler.scheduleAtFixedRate(new Runnable() {
+			@Override
+			public void run() {
+				try {
+					if(properlyConfigured){
+						execute();
+					}
+				} catch (Exception e) {
+					logger.debug("Exception during poll : {}", e);
+				}
+			}
+		}, 0, refresh, TimeUnit.SECONDS);
+
+	}
+
+	/**
+	 * Stops this thing's polling future
+	 */
+	private void stopPolling() {
+		if (pollFuture != null && !pollFuture.isCancelled()) {
+			pollFuture.cancel(true);
+			pollFuture = null;
+		}
+	}
+
+	/**
+	 * The polling future executes this every iteration
+	 */
+	protected void execute() {
+		logger.trace("Connecting to {}" + baseURL);
+		
+		clearState();
+		
+		// we will reconstruct the document with all the responses combined for
+		// XPATH
+		StringBuilder sb = new StringBuilder("<response>");
+
+		// pull down the three xml documents
+		String[] statuses = { "status", "chem", "pumps" };
+		for (String status : statuses) {
+			String response = getUrl(baseURL + "/" + status + ".xml", TIMEOUT);
+			logger.trace(baseURL + "/" + status + ".xml \n {}", response);
+			if (response == null) {
+				logger.error("No response from Autelis controller!");
+				return;
+			}
+			// get the xml data between the response tags and append to our main
+			// doc
+			Matcher m = responsePattern.matcher(response);
+			if (m.find()) {
+				sb.append(m.group(1));
+			}
+		}
+		// finish our "new" XML Document
+		sb.append("</response>");
+
+		/*
+		 * This xmlDoc will now contain the three XML documents we retrieved
+		 * wrapped in response tags for easier querying in XPath.
+		 */
+		String xmlDoc = sb.toString();
+		for (Channel channel : getThing().getChannels()) {
+			String key = channel.getUID().getId().replace('-', '/');
+			XPathFactory xpathFactory = XPathFactory.newInstance();
+			XPath xpath = xpathFactory.newXPath();
+			try {
+				InputSource is = new InputSource(new StringReader(xmlDoc));
+				String value = xpath.evaluate("response/" + key, is);
+				
+				State state = toState(channel.getAcceptedItemType(), value);
+				State oldState = stateMap.put(channel.getUID().getAsString(), state);
+				if(!state.equals(oldState)){
+					logger.debug("updating channel {} with state {}", channel, state);
+					updateState(channel.getUID(), state);
+				}
+			} catch (XPathExpressionException e) {
+				logger.error("could not parse xml", e);
+			}
+		}
+	}
+
+	/**
+	 * Simple logic to perform a authenticated GET request
+	 * @param url
+	 * @param timeout
+	 * @return
+	 */
+	private String getUrl(String url, int timeout) {
+		HttpClient client = new HttpClient();
```

Comment:
Using a RasPi1, I learned that instantiating an HttpCient is a pretty expensive operation. It is therefore much more efficient, to keep the client (with set credentials and other configuration) in a field within the class and thus reuse it on every new call.